src/prefect/cli/cloud.py

```diff
@@ -150,13 +299,81 @@ async def login(
     Creates a new profile configured to use the specified PREFECT_API_KEY.
     Uses a previously configured profile if it exists.
     """
+    if not is_interactive() and (not key or not workspace_handle):
```

Comment:
```suggestion
    needs_key_and_workspace = not is_interactive() and (not key or not workspace_handle)
    if needs_key_and_workspace:
```

Reply:
Yeah? I find this pretty clear as-is. It's not "if needs key and workspace" it's "if not interactive, let's throw an error if we'll need to ask for things"

Reply:
I'm flexible on this one - not strong feelings